osquery/events/windows/windows_event_log.h

```diff
@@ -0,0 +1,118 @@
+ * Subscribers can pass a vector of source values indicating which
+ * Windows event logs the subscriber wishes to subscribe to.
+ */
+struct WindowsEventLogSubscriptionContext : public SubscriptionContext {
+  /// Channel or Path of the Windows Event Log to subscribe to
+  std::set<std::wstring> sources;
+
+ private:
+  friend class WindowsEventLogEventPublisher;
+};
+
+/**
+ * @brief Event details for WindowsEventLogEventPublisher events.
+ *
+ * It is the responsibility of the subscriber to understand the best
+ * way in which to parse the event data. The publisher will convert the
+ * Event Log record into a boost::property_tree, and return the tree to
+ * the subscriber for further parsing and row population.
+ */
+struct WindowsEventLogEventContext : public EventContext {
+  /// A Windows event log record converted from XML
+  pt::ptree eventRecord;
+
+  /*
+   * In Windows event logs, the source to which an event belongs is referred
+   * to as the 'channel'. We keep track of the channel for each event, as the
+   * subscriber can decide to receive only events for specified channels.
+   */
+  std::wstring channel;
+};
+
+using WindowsEventLogEventContextRef =
+    std::shared_ptr<WindowsEventLogEventContext>;
+using WindowsEventLogSubscriptionContextRef =
+    std::shared_ptr<WindowsEventLogSubscriptionContext>;
+
+/**
+ * @brief A Windows Event Log Publisher
+ *
+ * This EventPublisher allows EventSubscriber's to subscribe to Windows
+ * Event Logs. By default we subscribe to all of the Windows system Event
+ * Log channels, and make _no_ filter queries on the events returned by
+ * the system, as any desired filtering should be handled at through SQL
+ * queries.
+ */
+class WindowsEventLogEventPublisher
+    : public EventPublisher<WindowsEventLogSubscriptionContext,
+                            WindowsEventLogEventContext> {
+  DECLARE_PUBLISHER("windows_event_log");
+
+ public:
+  /// Checks to see if a Event Log channel matches a given subscriber
+  bool shouldFire(const WindowsEventLogSubscriptionContextRef& mc,
+                  const WindowsEventLogEventContextRef& ec) const override;
+
+  void configure() override;
+
+  void tearDown() override;
+
+  /// The calling for beginning the thread's run loop.
+  Status run() override;
+
+  /// Windows Event Callback required for API calls
+  static unsigned long __stdcall winEventCallback(
+      EVT_SUBSCRIBE_NOTIFY_ACTION action, PVOID pContext, EVT_HANDLE hEvent);
+
+  /// Helper function to convert an XML event blob into a property tree
+  static Status WindowsEventLogEventPublisher::parseEvent(EVT_HANDLE evt,
+                                                          pt::ptree& propTree);
+
+ private:
+  /// Restarts the osquery Windows Event Log Events publisher
+  void restart();
+
+  /// Ensures that all Windows event log subscriptions are removed
+  void stop() override;
+
+  /// Returns whether or not the publisher has active subscriptions
+  bool isSubscriptionActive() {
```

Comment:
Even though this is a small function, should it's implementation be in the cpp file?